README.md

```diff
@@ -214,6 +224,21 @@ const linter = new Linter(fileName, fileContents, options);
 const result = linter.lint();
 ```
 
+#### Type Checking
+
+To enable rules that work with the type checker, a TypeScript program object may be passed to the linter. Helper functions are provided to create a program from a `tsconfig.json` file. A project directory can be specified if project files do not lie in the same directory as the `tsconfig.json` file.
```

Comment:
This should probably be edited to "... a TypesScript program [must] be passed to the linter [when using the programmatic API]." (my additions in brackets)

Also it would be good to clarify that a program is automatically created for you when using the appropriate CLI flags.